Sources/NIO/Bootstrap.swift

```diff
@@ -325,6 +325,47 @@ private extension Channel {
     }
 }
 
+/// `NIOClientTransportBootstrap` is implemented by various underlying transport mechanisms. Typically,
+/// this will be the BSD Sockets API implemented by `ClientBootstrap`.
+public protocol NIOClientTransportBootstrap {
```

Comment:
naming nit: maybe `NIOTCPClientBootstrap`? Because this bootstrap will only work for TCP stuff. But not strong on this one, @Lukasa ?

Reply:
I might prefer NIOStreamClientBootstrap, to enable the possibility of encapsulating other things.